python/ray/tune/ray_trial_executor.py

```diff
@@ -164,21 +92,41 @@ def noop_logger_creator(config, logdir):
     return NoopLogger(config, logdir)
 
 
+# The class responsible for communicating back to TrialRunner.
+class FutureResult:
+    def __init__(self, type, trial=None, result=None):
+        self.type = type
+        self.trial = trial
+        self.result = result
+
+    def __repr__(self):
+        return f"[{self.type}] for {self.trial}"
+
+
 @DeveloperAPI
 class RayTrialExecutor(TrialExecutor):
     """An implementation of TrialExecutor based on Ray."""
 
+    # Types of different futures handled by the executor.
+    NO_RUNNING_TRIAL_TIMEOUT = "NO_RUNNING_TRIAL_TIMEOUT"
+    PG_READY = "PG_READY"
+    TRAINING_RESULT = "TRAINING_RESULT"
+    SAVING_RESULT = "SAVING_RESULT"
+    RESTORING_RESULT = "RESTORING_RESULT"
+    STOP_RESULT = "STOP_RESULT"  # Internally to executor only.
+    # This is to signal to TrialRunner that there is an error.
+    ERROR = "ERROR"
+    PUNT = "PUNT"  # punting back to TrialRunner's main event loop.
+
```

Comment:
Should those be an Enum? Would make type checking and validation easier. Alternatively, an enum could be defined outside of the class while the attributes here use its values.